app/views/contexts/hub_context.py

```diff
@@ -49,19 +35,37 @@ class HubContext(Context):
     }
 
     def get_context(self, survey_complete, enabled_section_ids) -> Mapping:
-        context = self.HUB_CONTENT_STATES[
-            "COMPLETE" if survey_complete else "INCOMPLETE"
-        ]
-        context["rows"] = self._get_rows(enabled_section_ids)
+        hub_context = self._schema.get_hub()
+        rows = self._get_rows(enabled_section_ids)
+        custom_text = hub_context.get(
+            "complete" if survey_complete else "incomplete", {}
+        )
 
-        return context
+        if survey_complete:
+            return {
+                "title": custom_text.get("title") or lazy_gettext("Submit survey"),
+                "guidance": custom_text.get("guidance")
+                or lazy_gettext("Please submit this survey to complete it"),
+                "submit_button": hub_context.get("submission", {}).get("button")
+                or lazy_gettext("Submit survey"),
+                "rows": rows,
+            }
+
+        return {
+            "title": lazy_gettext("Choose another section to complete"),
+            "guidance": custom_text.get("guidance")
+            or lazy_gettext(
+                "You must complete all sections in order to submit this survey"
+            ),
+            "submit_button": lazy_gettext("Continue"),
+            "rows": rows,
+        }
```

Comment:
Not a blocker, but personally think it is a bit clearer to assign any changing vars then return.
```py
        if survey_complete:
            title = custom_text.get("title") or lazy_gettext("Submit survey")
            guidance = custom_text.get("guidance") or lazy_gettext(
                "Please submit this survey to complete it"
            )

            submit_button = (
                hub_context.get("submission", {}).get("button")
                or lazy_gettext("Submit survey"),
            )

        else:
            title = lazy_gettext("Choose another section to complete")
            guidance = custom_text.get("guidance") or lazy_gettext(
                "You must complete all sections in order to submit this survey"
            )
            submit_button = lazy_gettext("Continue")

        return {
            "title": title,
            "guidance": guidance,
            "submit_button": submit_button,
            "rows": rows,
        }
```

Reply:
`rows` also doesn't need to be assigned to a variable if it's done that way.